pipelines/ApPipeMultiTractFakes.yaml

```diff
@@ -0,0 +1,91 @@
+      doSubSelectSources: True
+      select_col: 'isTemplateSource'
+  processVisitFakes:
+    class: lsst.pipe.tasks.processCcdWithFakes.ProcessCcdWithVariableFakesTask
+    config:
+      connections.coaddName: goodSeeing
+      insertFakes.doSubSelectSources: True
+      insertFakes.select_col: 'isVisitSource'
+      calibrate.photoCal.match.referenceSelection.magLimit.fluxField: i_flux
+      calibrate.photoCal.match.referenceSelection.magLimit.maximum: 22.0
+  retrieveTemplate:  # New task added for multi-tract difference imaging
+    class: lsst.ip.diffim.getTemplate.GetMultiTractCoaddTemplateTask
+    config:
+      connections.coaddExposures: goodSeeingCoadd
+      connections.outputExposure: fakes_goodSeeingDiff_templateExp
+  imageDifference:  # New task class for multi-tract difference imaging
+    class: lsst.pipe.tasks.imageDifference.ImageDifferenceFromTemplateTask
+    config:
+      connections.fakesType: 'fakes_'
+      connections.exposure: fakes_calexp
+      connections.outputSchema: fakes_goodSeeingDiff_diaSrc_schema
+      connections.subtractedExposure: fakes_goodSeeingDiff_differenceExp
+      connections.scoreExposure: fakes_goodSeeingDiff_scoreExp
+      connections.warpedExposure: fakes_goodSeeingDiff_warpedExp
+      connections.matchedExposure: fakes_goodSeeingDiff_matchedExp
+      connections.diaSources: fakes_goodSeeingDiff_diaSrc
+      connections.inputTemplate: fakes_goodSeeingDiff_templateExp
+      # The following are copied from ApPipe.yaml (imageDifference class changed)
+      doWriteWarpedExp: True  # Required for packaging alerts in diaPipe
+      doSkySources: True
+      coaddName: goodSeeing
+      getTemplate.coaddName: goodSeeing
+      connections.coaddName: goodSeeing
+  transformDiaSrcCat:
+    class: lsst.ap.association.TransformDiaSourceCatalogTask
+    config:
+      connections.fakesType: 'fakes_'
+      connections.diaSourceSchema: fakes_goodSeeingDiff_diaSrc_schema
+      connections.diaSourceCat: fakes_goodSeeingDiff_diaSrc
+      connections.diffIm: fakes_goodSeeingDiff_differenceExp
+      connections.diaSourceTable: fakes_goodSeeingDiff_diaSrcTable
+  diaPipe:
+    class: lsst.ap.association.DiaPipelineTask
+    config:
+      doWriteAssociatedSources: True
+      connections.fakesType: 'fakes_'
+      connections.diaSourceTable: fakes_goodSeeingDiff_diaSrcTable
+      connections.diffIm: fakes_goodSeeingDiff_differenceExp
+      connections.warpedExposure: fakes_goodSeeingDiff_warpedExp
+      connections.associatedDiaSources: fakes_goodSeeingDiff_assocDiaSrc
+      # apdb.db_url: YOU MUST CONFIGURE THIS BEFORE RUNNING THE PIPELINE
+  fakesMatch:
+    class: lsst.pipe.tasks.matchFakes.MatchFakesTask
+    config:
+      connections.coaddName: goodSeeing
+      matchDistanceArcseconds: 0.5
+  fracDiaSourcesToSciSources:
+    class: lsst.ip.diffim.metrics.FractionDiaSourcesToSciSourcesMetricTask
+    config:
+      connections.fakesType: 'fakes_'
+      connections.coaddName: goodSeeing
```

Comment:
An interesting point! Since this is a fakes pipeline, I think this metric task is close to essential. How about I comment it out and users can decide, and then it doesn't interfere with future ap_verify imports.

Reply:
I'm confused. Why is it essential?

Reply:
Essential scientifically, not programmatically... why run fakes if you're not going to make matched fakes catalogs?

Reply:
Isn't that `MatchFakesTask`? `FractionDiaSourcesToSciSourcesMetricTask` just computes a statistic, and doesn't even know about fakes.

Reply:
I swore this was for fakes, but you have a valid point. Maybe it used to be for fakes? Hmm

Reply:
I'll get rid of it 😄